### DISCUSSION TITLE
FindMarkers reporting different genes in v5 / backwards compatibility of seurat object?

### DISCUSSION THREAD

**Discussion by mahogny** [satijalab/seurat#8229] [Q&A] [answered]
Hi!

We were recently forced to update to Seurat v5 from v4 or v3 -- unfortunately not clear which version, and this doesn't help the troubleshooting

Now FindAllMarkers report very different genes from before, to the extent that I question if this is random or not (based on plotting genes on UMAPs). It's hard to go back and do a proper comparison unfortunately

Has anything been changed here that could affects things? We use default settings for this function. Might it be using the wrong data layer or something?

Notably: We are loading an old seurat RDS with the new version. I imagine tons can go wrong here. But much processed data published out there /is/ in the old format, so I think seurat will need a sensible route to update such old objects and be backwards compatible. Is this the plan? I think some general recommendation might be due in terms of how we should release processed data in the future (besides just raw counts and our code)

Best,
Johan

**Comment by CMUltyao** [comment]
And also, V5 seems generate much higher fold change and more DEGs than V4. which setting is different?

**Reply by Shuresearcher** [reply]
Do you know why is it? Can we trust DEGs from V5?

**Comment by Yanst86** [comment, marked as answer]
They switched the algorithm for differential gene expression - https://satijalab.org/seurat/articles/announcements.html
In addition, please note that for FindAllMarkers they changed the min.pct threshold to 0.01 instead of 0.1 - that's why you also get genes that are expressed in a very low fraction of the cells that you didn't see at all in Seurat V4.

**Reply by mahogny** [reply]
Great answer, thanks! It probably would be best if the previous min.pct was kept for backwards compatibility!

**Reply by Shuresearcher** [reply]
Does this changed affect fold change of DEGs？

**Reply by sarahhp** [reply]
I noticed the algorithm change (documentated in help), but not the min.pct! Thanks

**Reply by sarahhp** [reply]
Also logfc.threshold changed from 0.25 to 0.1 with v5.

**Comment by nicolaromano** [comment]
Having just stumbled in this same problem, one solution (if you know which version you were using) is to use `install_version` from the `remotes` package and do, for instance

```
 install_version("Seurat", "4.3.0")
```

Using something like `renv` might also help

**Comment by Elo-mars** [comment]
Personally, I use Seurat v5 for ALL my analyses, except FindAllMarkers (or FindMarkers) - for that, I switch my conda environment and use Seurat 4. we saw that the DEG are much better-biologically make sense- with the previous Seurat version

**Comment by Elo-mars** [comment]
Is there a better DEG calculation? 
can someone say it is better with Seurat 4 or Seurat 5? (I know the calculations are different, the log2FC is also very different - before log2FC of 1 meant  2^1=2, so the gene was twice as more expressed in group 1 than 2, what does it mean now? and I know it changed because now they pseudo-bulk all, so it is much quicker) but the results are quite different. (even if I say min % is 10% for both version) - 

I do not know what to think anymore ... help?  

quid for 10X data, but also quid of smartseq2 data?  (I know that you can use test=wilcox_limma to normally reproduce the results of S4, does that mean they think S4 was better for DEG???)

**Reply by samuel-marsh** [reply]
Personally for true DE analysis (between groups, treatments, etc) I use pseudo bulk and/or other methods directly both because you have more control (e.g. being able to use both fixed and random effects in MAST, other parameters in DESeq2, limma etc).  

I use Seurat’s DE for marker genes when clustering (where p values as discussed aren’t really valid) but where log2FC and pct difference in expression can be valuable. 

the speed up in results is due to implementation of Wilcox test using presto package which dramatically improves for large datasets.

log2FC values still mean the same thing in terms of values.  1 is two fold up, ~0.58 is 1.5 fold etc.  As mentioned in release notes the pseudo count change is only difference which is why the values of the log2FC differ between versions.  As mentioned this can be issue for lowly expressed genes but overall the way pseudo count is now applied is more appropriate for current state of the field.

in terms of Smart-seq data the appropriateness will depend on several factors including whether or not data has UMIs.

Hope that helps!
Best,
Sam
